dsl/src/main/scala/org/http4s/dsl/impl/Path.scala

```diff
@@ -166,10 +166,13 @@ object /: {
     }
 }
 
-protected class PathVar[A](cast: String => Try[A]) {
+protected class PathVar[A](cast: String => Option[A]) {
+  def this(cast: String => Try[A])(implicit ev: DummyImplicit) =
+    this((str: String) => cast(str).toOption)
+
   def unapply(str: String): Option[A] =
-    if (!str.isEmpty)
-      cast(str).toOption
+    if (str.nonEmpty)
```

Comment:
I'm not sure why this non-empty check is baked in, but it always has been, so I guess it's right for semantic compatibility.